include/revng/PipelineDescriptor/AnalysisArgument.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+/* TUPLE-TREE-YAML
+name: AnalysisArgument
```

Comment:
This seems to be unused, but there's a fixup later on. Backport it.

Also, these are the accepted kinds, they don't have anything to do with analysis arguments.

The anlaysis arguments are options, but the accepted kinds is a property of the analysis, not of the argument.

Reply:
> Backport it

Will do

> Also, these are the accepted kinds, they don't have anything to do with analysis arguments.

There are 2 different things:
* An analysis argument, which needs to point to a target in the pipeline to produce a result (e.g. `ImportBinary` has an argument `input` which accepts kind `Binary`)
* An analysis option, which (AFAIK) did not have any API in PipelineC to introspect them since (at the time of writing the API) no analysis had them

Reply:
Changed the name to `AnalysisContainerInput`, also added Options